editor/js/Sidebar.Material.js

```diff
@@ -680,189 +688,224 @@ Sidebar.Material = function ( editor ) {
+		}
+
+		if ( material.envMap !== undefined ) {
+
+			materialEnvMapEnabled.setValue( material.envMap !== null );
+
+			if ( material.envMap !== null || resetTextureSelectors ) {
 
-				materialEnvMapEnabled.setValue( material.envMap !== null );
 				materialEnvMap.setValue( material.envMap );
-				materialReflectivity.setValue( material.reflectivity );
 
 			}
 
-			if ( material.lightMap !== undefined ) {
+			materialReflectivity.setValue( material.reflectivity );
+
+		}
+
+		if ( material.lightMap !== undefined ) {
+
+			materialLightMapEnabled.setValue( material.lightMap !== null );
+
+			if ( material.lightMap !== null || resetTextureSelectors ) {
 
-				materialLightMapEnabled.setValue( material.lightMap !== null );
 				materialLightMap.setValue( material.lightMap );
 
 			}
 
-			if ( material.aoMap !== undefined ) {
+		}
+
+		if ( material.aoMap !== undefined ) {
+
+			materialAOMapEnabled.setValue( material.aoMap !== null );
+
+			if ( material.aoMap !== null || resetTextureSelectors ) {
 
-				materialAOMapEnabled.setValue( material.aoMap !== null );
 				materialAOMap.setValue( material.aoMap );
-				materialAOScale.setValue( material.aoMapIntensity );
 
 			}
 
-			if ( material.side !== undefined ) {
+			materialAOScale.setValue( material.aoMapIntensity );
 
-				materialSide.setValue( material.side );
+		}
 
-			}
+		if ( material.side !== undefined ) {
 
-			if ( material.shading !== undefined ) {
+			materialSide.setValue( material.side );
 
-				materialShading.setValue( material.shading );
+		}
 
-			}
+		if ( material.shading !== undefined ) {
 
-			if ( material.blending !== undefined ) {
+			materialShading.setValue( material.shading );
 
-				materialBlending.setValue( material.blending );
+		}
 
-			}
+		if ( material.blending !== undefined ) {
 
-			if ( material.opacity !== undefined ) {
+			materialBlending.setValue( material.blending );
 
-				materialOpacity.setValue( material.opacity );
+		}
 
-			}
+		if ( material.opacity !== undefined ) {
 
-			if ( material.transparent !== undefined ) {
+			materialOpacity.setValue( material.opacity );
+
+		}
 
-				materialTransparent.setValue( material.transparent );
+		if ( material.transparent !== undefined ) {
 
-			}
+			materialTransparent.setValue( material.transparent );
 
-			if ( material.wireframe !== undefined ) {
+		}
 
-				materialWireframe.setValue( material.wireframe );
+		if ( material.wireframe !== undefined ) {
 
-			}
+			materialWireframe.setValue( material.wireframe );
 
-			if ( material.wireframeLinewidth !== undefined ) {
+		}
 
-				materialWireframeLinewidth.setValue( material.wireframeLinewidth );
+		if ( material.wireframeLinewidth !== undefined ) {
 
-			}
+			materialWireframeLinewidth.setValue( material.wireframeLinewidth );
+
+		}
 
-			updateRows();
+		setRowVisibility();
+
+	}
+
+	// events
+
+	signals.objectSelected.add( function ( object ) {
+
+		if ( object && object.material ) {
+
+			var objectChanged = object !== currentObject;
```

Comment:
As can be seen in the last commit; 'objectChanged' is passed as the actual argument to 'refreshUi(resetTextureSelectors)'.

Temporarily disabled texture images are stored solely in the GUI controls. When disabled, the uniform value is set to 'null'. Reading back that state into the GUI lets the texture images fade into oblivion. This behavior is only desired when loading another object's material into the sidebar, not when reacting to changes of the material (from the script editor, or because a new material object was created changing the material class).